packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -129,63 +272,71 @@ contract FeeHandler is Ownable, Initializable, UsingRegistry, ICeloVersionedCont
   }
 
   function _setDailyBurnLimit(address token, uint256 newLimit) private {
-    dailyBurnLimit[token] = newLimit;
+    TokenState storage tokenState = tokenStates[token];
+    tokenState.dailyBurnLimit = newLimit;
     emit DailyLimitSet(token, newLimit);
   }
 
-  /**
-    * @notice Allows owner to set the router for a token.
-    * @param token Address of the token to set.
-    * @param router The new router.
-    */
-  function setRouter(address token, address router) external onlyOwner {
-    _setRouter(token, router);
+  function burnCelo() external {
+    return _burnCelo();
   }
 
-  function _setRouter(address token, address router) private {
-    require(router != address(0), "Router can't be address zero");
-    routerAddresses[token].push(router);
-    emit RouterAddressSet(token, router);
+  function handleAll() external {
+    return _handleAll();
   }
 
-  /**
-    * @notice Allows owner to remove a router for a token.
-    * @param token Address of the token.
-    * @param router Address of the router to remove.
-    * @param index The index of the router to remove.
-    */
-  function removeRouter(address token, address router, uint256 index) external onlyOwner {
-    require(routerAddresses[token][index] == router, "Index does not match");
+  function _handleAll() private {
+    for (uint256 i = 0; i < activeTokens.length; i++) {
+      // calling _handle will trigger a lot of burn Celo that can be just batched at the end
+      // _handle(activeTokens[i]);
+      _sell(activeTokens[i]);
+      _distribute(activeTokens[i]); // TODO move to _distributeAll()
+    }
+    _burnCelo();
+  }
 
-    uint256 length = routerAddresses[token].length;
-    routerAddresses[token][index] = routerAddresses[token][length - 1];
-    routerAddresses[token].pop();
-    emit RouterAddressRemoved(token, router);
+  function handle(address tokenAddress) external {
+    return _handle(tokenAddress);
   }
 
-  /**
-    * @notice Get the list of routers for a token.
-    * @param token The address of the token to query.
-    * @return An array of all the allowed router.
-    */
-  function getRoutersForToken(address token) external view returns (address[] memory) {
-    return routerAddresses[token];
+  function _handle(address tokenAddress) private {
+    // Celo doesn't have to be exchanged for anything
+    if (tokenAddress != registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID)) {
+      _sell(tokenAddress);
+    }
+    _burnCelo();
+    _distribute(tokenAddress);
   }
 
   /**
     * @notice Burns all the Celo balance of this contract.
     */
-  function burnAllCelo() public {
+  function _burnCelo() private {
+    TokenState storage tokenState = tokenStates[registry.getAddressForOrDie(
```

Comment:
If we are relying on Celo being in tokenStates we should probably require it being set in constructor.